### PR TITLE
Switch to high risk maternity careplan

### PR DESCRIPTION
**Pull request by vinay-debug**
### Related issue
  - *Issue Id*
  - *Context*
  
### List of changes
  - *change a*
  - *change b*

### Database migration/schema changes (if any)
- *List your migrations here...*

### Self review checklist
  - [ ] Confirm that the changes made by you meet the resolution criteria and nothing else.
  - [ ] Ran all the tests locally to check that you have not introduced any new regression.
  - [ ] If there are any failing tests, you have already fixed them. If not 
  - [ ] Followed coding and styling guidelines.
  - [ ] Made sure the changes are is not overly complex, unreadable and unnecessarily smart. Simplicity wins!
  - [ ] Made sure the comments are not too verbose and not to cryptic.
  - [ ] Made sure the code metrics are not deteriorated.
  - [ ] Made sure you have updated (if necessary)-
    - [ ] Documentation
    - [ ] Translation
    - [ ] Tests/test data
    - [ ] Postman collection
    - [ ] Version
       
### Additional info (if any)
- *Add anything you feel worth mentioning...*

### screensshots (if any)

Original | Updated
:------------------------:|-------------------------------|
** Original screenshot ** | ** Updated screenshot **|

### CODE REVIEW

File: src/database/sql/sequelize/repositories/clinical/careplan/careplan.repo.ts [hunk 1]
```diff
@@ -391,8 +406,7 @@ export class CareplanRepo implements ICareplanRepo {
 
             const foundResults = await CareplanActivity.findAndCountAll({
                 where : {
-                    Provider : "REAN",
-                    PlanName : "Maternity Careplan"
+                    Provider : "REAN"
```
Comment (kiran-rean) [inline src/database/sql/sequelize/repositories/clinical/careplan/careplan.repo.ts:409]: Why are you hard-coding Provider as REAN?

File: src/database/sql/sequelize/repositories/clinical/careplan/careplan.repo.ts [hunk 2]
```diff
@@ -408,4 +422,61 @@ export class CareplanRepo implements ICareplanRepo {
...
+            await updateRisk.save();
+
+            return EnrollmentMapper.toDto(updateRisk);
+        } catch (error) {
+            Logger.instance().log(error.message);
+            throw new ApiError(500, error.message);
+        }
+    };
+
+    public deleteFutureCareplanTask = async (enrollment: EnrollmentDomainModel): Promise<number> => {
```
Comment (kiran-rean) [inline src/database/sql/sequelize/repositories/clinical/careplan/careplan.repo.ts:451]: Change the name from 'deleteFutureCareplanTask' to 'deleteFutureCareplanTasks'.

File: src/modules/careplan/providers/rean/rean.decision.making.service.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+@injectable()
+export class HighRiskCareplanService {
+
+    _handler: CareplanHandler = new CareplanHandler();
+
+    constructor(
+        @inject('ICareplanRepo') private _careplanRepo: ICareplanRepo
+    ) {}
+
+    deleteFutureCareplanTask = async(enrollment: EnrollmentDto): Promise<number> => {
```
Comment (kiran-rean) [inline src/modules/careplan/providers/rean/rean.decision.making.service.ts:18]: change the function name here also.

File: src/services/clinical/careplan.service.ts [hunk 4]
```diff
@@ -575,6 +584,65 @@ export class CareplanService implements IUserActionService {
...
+        const patient = await this._patientRepo.search(filter);
+        if (patient.Items.length === 0) {
+            throw new Error('Patient does not exist!');
+        }
+        updateRisk.PatientUserId = patient.Items[0].UserId;
+
+        return  await this._careplanRepo.updateRisk(updateRisk);
+    };
+
+    public scheduleDailyHighRiskCareplan = async (): Promise<void> => {
```
Comment (kiran-rean) [inline src/services/clinical/careplan.service.ts:601]: This function should not be part of the careplan.service.ts. 
It should be moved to the plugins / custom handlers.

**Review by kiran-rean — DISMISSED** [review]

File: src/database/sql/sequelize/repositories/clinical/careplan/careplan.repo.ts (hunk 1 above)
Comment (vinay-debug) [reply src/database/sql/sequelize/repositories/clinical/careplan/careplan.repo.ts:409]: Done

File: src/services/clinical/careplan.service.ts (hunk 4 above)
Comment (vinay-debug) [reply src/services/clinical/careplan.service.ts]: Done

File: src/database/sql/sequelize/repositories/clinical/careplan/careplan.repo.ts (hunk 2 above)
Comment (vinay-debug) [reply src/database/sql/sequelize/repositories/clinical/careplan/careplan.repo.ts:451]: Done

**Review by tabbasum-rean — APPROVED** [review]

**Review by tabbasum-rean — APPROVED** [review]

**Pull request merged**